agent/consul/leader.go

```diff
@@ -626,8 +626,8 @@ func (s *Server) handleDeregisterMember(reason string, member serf.Member) error
 
 // joinConsulServer is used to try to join another consul server
 func (s *Server) joinConsulServer(m serf.Member, parts *metadata.Server) error {
-	// Do not join ourself
-	if m.Name == s.config.NodeName {
+	// Do not join ourself if we are the only member
```

Comment:
If I removed this check entirely, [TestAgent_CheckAdvertiseAddrsSettings](https://github.com/hashicorp/consul/blob/831d84c940f4aec365973bd4e35eb891bdf1b732/agent/agent_test.go#L109-L109) failed. Seems like it was trying to join the same node id with a different IP address, and if this check went away entirely it would keep removing and adding itself and never finishing leader election.